vision/standard_object/odlc_classify_shape.py

```diff
@@ -69,420 +95,283 @@ def process_shapes(
+                y, x = point
+                max_y = max(y, max_y)
+                min_y = min(y, min_y)
+                max_x = max(x, max_x)
+                min_x = min(y, min_y)
+            height: int = max_y - min_y
+            width: int = max_x - min_x
+            vertices = tlwh_to_vertices(min_x, max_y, width, height)
+
+        bounding_box: bbox
+        bbox.__init__(bounding_box, vertices, ObjectType.STD_OBJECT, None)
+        bbox_list.append(bounding_box)
+    return bbox_list
+
+
+def classify_shape(contour: consts.Contour) -> chars.ODLCShape | None:
     """
-    Will first determine whether the specified contour is an ODLC shape, then will determine
-    which ODLC shape it is.
+    Will determine if the contour matches any ODLC shape, then verify that choice by comparing to sample
 
     Parameters
     ----------
     contour : consts.Contour
         A contour returned by the contour detection algorithm
-    image_dims : tuple[int, int]
-        The dimensions of the original entire image
-        (only height and width as gotten from image.shape[:2], not the color channels)
-        image_dim_height : int
-            The height dimension of the image
-        image_dim_width : int
-            The width dimension of the image
-    approx_contour : consts.Contour | None = None
-        Optional parameter to provide the approximated version (from cv2.approxPolyDP) of the
-        contour to be checked to avoid recalculation
-        This is None by default, it is only not None when the approximated contour is provided
 
     Returns
     -------
     shape : chars.ODLCShape | None
         Will return one of the ODLC shapes defined in vision/common/odlc_characteristics or None
-        if the given contour is not an ODLC shape (fails filtering or doesnt match any ODLC)
+        if the given contour is not an ODLC shape (doesn't match any ODLC)
     """
-    if approx_contour is None:
-        approx_contour = cv2.approxPolyDP(contour, APPROX_CNT_EPSILON, True)
-
-    is_shape: bool
-    is_circular: bool
-    is_shape, is_circular = filtering.filter_contour(contour, image_dims, approx_contour)
-
-    if not is_shape:
-        return None
-
-    defects: int = check_concavity(approx_contour)
-
-    shape: chars.ODLCShape | None
-    if defects not in {0, 4, 5}:
-        shape = None  # if has not 0, 4, or 5 defects then not a valid shape
-    elif is_circular and defects == 0:
-        shape = classify_circular(contour)
-    else:
-        shape = check_polygons(approx_contour)
-
-        if (defects == 4) ^ (shape == chars.ODLCShape.CROSS):  # logical xor
-            shape = None  # one of above cannot be true while the other is false if valid shape
-        elif (defects == 5) ^ (shape == chars.ODLCShape.STAR):
-            shape = None  # one of above cannot be true while the other is false if valid shape
-
-    return shape
+    polar_array: PolarArray = generate_polar_array
```

Comment:
I assume this should be a function call
```suggestion
    polar_array: PolarArray = generate_polar_array(contour)
```